crawler4j-core/src/main/java/edu/uci/ics/crawler4j/crawler/WebCrawler.java

```diff
@@ -236,24 +233,23 @@ protected void onUnexpectedStatusCode(String urlStr, int statusCode, String cont
      * This function is called if the content of a url could not be fetched.
      *
      * @param webUrl URL which content failed to be fetched
-     * @deprecated use {@link #onContentFetchError(Page)}
      */
-    @Deprecated(forRemoval = true)
     protected void onContentFetchError(WebURL webUrl) {
         logger.warn("Can't fetch content of: {}", webUrl.getURL());
         // Do nothing by default (except basic logging)
         // Sub-classed can override this to add their custom functionality
     }
-
+    
     /**
-     * This function is called if the content of a url could not be fetched.
+     * This function is called if there has been an error in parsing the content.
      *
-     * @param page Partial page object
+     * @param webUrl URL which failed on parsing
      */
-    protected void onContentFetchError(Page page) {
```

Comment:
I'll put it back in. Since this method is called when fetch goes wrong, all this incoming object contains is the result of `Page page = new Page(curURL);`, so basically the WebURL.

Reply:
Yup - with the removal flag, we can go for removal in the next versions :)